### PR TITLE
Generate: TF can now accept custom logits processors

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

TF generation test addition PR 1 (out of ???).

In an effort to move generation integration tests to be framework-agnostic, I'll be adding low-hanging fruit to TF. This PR brings custom logits processors to TF `.generate()`. The code added is almost copy-paste from PT.

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for working on this!
